mantidproject/mantid
# Add indeterminate progress bar below EngDiff FitPropertyBrowser

**Description of work.**
Add new indeterminate progress bar to show when fitting is happening or has finished on the EngDiff fitting tab.

**To test:**

Please try to break this new progress bar in as many ways as possible! And together we can check that its style is clear on all OS (I've checked Ubuntu and MacOS so far). It has a moving partially blue bar when fitting, solid green for successful fit and empty with red border for failed fit. The progress bar resets to empty with a normal grey border when the fit browser is toggled open/close or a workspace is added or removed from the plot. The progress bar should mostly be in sync with the status reported text at the top of the embedded fit property browser.

- Load some focussed data into the EngDiff fitting tab and add it to the plot. Click the Fit toolbar button to enable the FitPropertyBrowser. Right-click in the plot and add a peak. 
- Click Fit > Fit and notice how the progress bar works.
- Add a few more peaks to get a longer fitting time.
- When a peaks are added to the model you should be able to click "Sequential Fit" or "Serial Fit" in the data table just above the plot. Check these methods of fitting also interact with the progress bar well.
- When sequentially or serially fitting it is possible that some spectra may succeed and some may fail. The progress bar should be in sync with the reported status at the top of the Fit property browser and report the status for the last fitted spectrum!
- again please try and break this as much as possible!
- check if tests are good enough and if I need more, I'd really like this new feature to be reliable for the users!

Fixes #34414 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
